wagtail/wagtailcore/models.py

```diff
@@ -365,6 +365,14 @@ class Page(six.with_metaclass(PageBase, MP_Node, ClusterableModel, index.Indexed
         editable=False
     )
 
+    old_url = models.TextField(
```

Comment:
Instead of maintaining a DB field on `pre_save` and comparing it on `post_save` , could you set a self.\_original_url_path in Page's `__init__()` method and compare it in `post_save`? Based on this solution http://stackoverflow.com/a/36722021/2141313